[harvester doc] doc about how to expand pvc size (#212)
harvester/docs: docs/troubleshooting/monitoring.md

cattle-monitoring-system   prometheus-rancher-monitoring-prometheus-db-prometheus-rancher-monitoring-prometheus-0           Bound    pvc-7c6dcb61-51a9-4a38-b4c5-acaa11788978   50Gi       RWO            harvester-longhorn   43h
cattle-monitoring-system   rancher-monitoring-grafana                                                                       Bound    pvc-b2b2c07c-f7cd-4965-90e6-ac3319597bf7   2Gi        RWO            harvester-longhorn   43h

# kubectl get volume -A
NAMESPACE         NAME                                       STATE      ROBUSTNESS   SCHEDULED   SIZE          NODE     AGE
longhorn-system   pvc-1b2fbbe9-14b1-4a65-941a-7d5645a89977   attached   degraded                 5368709120    harv31   43h
longhorn-system   pvc-7c6dcb61-51a9-4a38-b4c5-acaa11788978   attached   degraded                 53687091200   harv31   43h
longhorn-system   pvc-b2b2c07c-f7cd-4965-90e6-ac3319597bf7   attached   degraded                 2147483648    harv31   43h
```

### Scale down deployment

> The `Volume` needs to be detached at first.

Reviewer: ```suggestion
To scale down a deployment, you need to detach the `Volume` first.
```

> Following example is against the PVC claimed by `rancher-monitoring-grafana`.

Reviewer: ```suggestion
The example below is against the PVC claimed by `rancher-monitoring-grafana`.
```

> Find out the `deployment` in namespace `cattle-monitoring-system`.

Reviewer: ```suggestion
Find out the `deployment` in the namespace `cattle-monitoring-system`.
```

```
harv31:~ # kk get deployment -n cattle-monitoring-system
NAME                                    READY   UP-TO-DATE   AVAILABLE   AGE
rancher-monitoring-grafana              1/1     1            1           43h
rancher-monitoring-kube-state-metrics   1/1     1            1           43h
rancher-monitoring-operator             1/1     1            1           43h
rancher-monitoring-prometheus-adapter   1/1     1            1           43h
```

Scale down the deployment `rancher-monitoring-grafana` to 0.

> `kubectl scale --replicas=0 deployment/rancher-monitoring-grafana  -n cattle-monitoring-system`

Reviewer: Please use ``` to wrap this line of code

> Check deployment and volume.

Reviewer: ```suggestion
Check the deployment and volume.
```

```
# kubectl get deployment -n cattle-monitoring-system
NAME                                    READY   UP-TO-DATE   AVAILABLE   AGE
rancher-monitoring-grafana              0/0     0            0           43h
rancher-monitoring-kube-state-metrics   1/1     1            1           43h
rancher-monitoring-operator             1/1     1            1           43h
rancher-monitoring-prometheus-adapter   1/1     1            1           43h

# kubectl get volume -A
NAMESPACE         NAME                                       STATE      ROBUSTNESS   SCHEDULED   SIZE          NODE     AGE
longhorn-system   pvc-1b2fbbe9-14b1-4a65-941a-7d5645a89977   attached   degraded                 5368709120    harv31   43h
longhorn-system   pvc-7c6dcb61-51a9-4a38-b4c5-acaa11788978   attached   degraded                 53687091200   harv31   43h
longhorn-system   pvc-b2b2c07c-f7cd-4965-90e6-ac3319597bf7   detached   unknown                  2147483648             43h
```

### Expand Volume

> In Longhorn WebUI, the related volume becomes `Detached`. Click `Operation` colume menu, and select `Expand Volume`.

Reviewer: ```suggestion
In the Longhorn WebUI, the related volume becomes `Detached`. Click the icon in the `Operation` column and select `Expand Volume`.
```

```

## Expand PV/Volume Size

`Harvester` integrates `Longhorn` as the default storage provider.

> Harvester `Monitoring` uses `Persistant Volume (PV)` to store running data. When a cluster has been running for a certain time, the `Persistant Volume` may need to expand its size.
> 

Reviewer: ```suggestion
Harvester `Monitoring` uses `Persistent Volume (PV)` to store running data. When a cluster has been running for a certain time, the `Persistent Volume` may need to expand its size.

```

Reviewer: I found that some previous suggesions haven't been applied but marked as resolved. Please double-check~

Based on the `Longhorn` `Volume` expansion guide, `Harvester` illustrates how to [expand the volume size](https://longhorn.io/docs/1.3.2/volumes-and-nodes/expansion/).

### View Volume

#### From Embedded Longhorn WebUI

> Access the embedded Longhorn WebUI according to [this document](https://docs.harvesterhci.io/v1.0/troubleshooting/harvester/#access-embedded-longhorn).

Reviewer 2: We could use a relative path here.

cattle-monitoring-system   prometheus-rancher-monitoring-prometheus-db-prometheus-rancher-monitoring-prometheus-0           Bound    pvc-7c6dcb61-51a9-4a38-b4c5-acaa11788978   50Gi       RWO            harvester-longhorn   43h
cattle-monitoring-system   rancher-monitoring-grafana                                                                       Bound    pvc-b2b2c07c-f7cd-4965-90e6-ac3319597bf7   2Gi        RWO            harvester-longhorn   43h

# kubectl get volume -A
NAMESPACE         NAME                                       STATE      ROBUSTNESS   SCHEDULED   SIZE          NODE     AGE
longhorn-system   pvc-1b2fbbe9-14b1-4a65-941a-7d5645a89977   attached   degraded                 5368709120    harv31   43h
longhorn-system   pvc-7c6dcb61-51a9-4a38-b4c5-acaa11788978   attached   degraded                 53687091200   harv31   43h
longhorn-system   pvc-b2b2c07c-f7cd-4965-90e6-ac3319597bf7   attached   degraded                 2147483648    harv31   43h
```

### Scale Down a Deployment

> To detach the `Volume`, you need to scale down the `deployment` which uses the `Volume`.

Author: Question: what's the difference between `that` and `which` here?

> To detach the `Volume`, you need to scale down the `deployment` which uses the `Volume`.

Reviewer: I think `that` is more grammatically correct ^^

The example below is against the PVC claimed by `rancher-monitoring-grafana`.

Find out the `deployment` in the namespace `cattle-monitoring-system`.

```
# kubectl get deployment -n cattle-monitoring-system
NAME                                    READY   UP-TO-DATE   AVAILABLE   AGE
rancher-monitoring-grafana              1/1     1            1           43h  // target deployment
rancher-monitoring-kube-state-metrics   1/1     1            1           43h
rancher-monitoring-operator             1/1     1            1           43h
rancher-monitoring-prometheus-adapter   1/1     1            1           43h
```

Scale down the deployment `rancher-monitoring-grafana` to 0.

```

> # kubectl scale --replicas=0 deployment/rancher-monitoring-grafana  -n cattle-monitoring-system

Reviewer 2: ```suggestion
# kubectl scale --replicas=0 deployment/rancher-monitoring-grafana -n cattle-monitoring-system
```
(nit).

rancher-monitoring-kube-state-metrics   1/1     1            1           43h
rancher-monitoring-operator             1/1     1            1           43h
rancher-monitoring-prometheus-adapter   1/1     1            1           43h

# kubectl get volume -A
NAMESPACE         NAME                                       STATE      ROBUSTNESS   SCHEDULED   SIZE          NODE     AGE
longhorn-system   pvc-1b2fbbe9-14b1-4a65-941a-7d5645a89977   attached   degraded                 5368709120    harv31   43h
longhorn-system   pvc-7c6dcb61-51a9-4a38-b4c5-acaa11788978   attached   degraded                 53687091200   harv31   43h
longhorn-system   pvc-b2b2c07c-f7cd-4965-90e6-ac3319597bf7   detached   unknown                  2147483648             43h  // volume is detached
```

### Expand Volume

In Longhorn WebUI, the related volume becomes `Detached`. Click the icon in the `Operation` colume, and select `Expand Volume`.

![](/img/v1.1/troubleshooting/4-select-volume-to-expand.png)

Input a new size, and `Longhorn` will expand the volume to this size.

![](/img/v1.1/troubleshooting/5-expand-volue-to-new-size.png)

### Scale Up a Deployment

> To scale up a deployment to the original replicas:

Reviewer: `to the original replicas` needs to be more detailed as mentioned before

Author: tried to add more details, please check again.

BTW: A common understanding is that the `trouble-shooting` is targeting those technical users, when they have certain background of `kubernetes`, they will not be confused by `replica` concept.

Reviewer: Can we say: To scale up a deployment to `1`:
I remember suggesting it in yesterday's comment, but I'm not sure if my understanding is correct

Author: The `scale up` is the reverse operation of `scale down`, it needs to keep the number of `replicas` as the original values. We use `the original replicas` to highlight this key point.

Please take a look of the updated description, thanks.

```

> # kubectl scale --replicas=1 deployment/rancher-monitoring-grafana  -n cattle-monitoring-system

Reviewer 2: ```suggestion
# kubectl scale --replicas=1 deployment/rancher-monitoring-grafana -n cattle-monitoring-system
```
ditto.
